src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1853,15 +1854,26 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
                 if (true !== $pool['tags'] && ($config['pools'][$pool['tags']]['tags'] ?? false)) {
                     $pool['tags'] = '.'.$pool['tags'].'.inner';
                 }
-                $container->register($name, TagAwareAdapter::class)
-                    ->addArgument(new Reference('.'.$name.'.inner'))
-                    ->addArgument(true !== $pool['tags'] ? new Reference($pool['tags']) : null)
-                    ->setPublic($pool['public'])
-                ;
+
+                $isRedis = $definition instanceof ChildDefinition && 'cache.adapter.redis' === $definition->getParent();
+                if (true === $pool['tags'] && $isRedis) {
+                    $definition = new Definition(RedisTagAwareAdapter::class, [null, null, 0, null]);
```

Comment:
There is a drawback to this approach: `RedisTagAwareAdapter` requires a specific eviction policy. This means this will break for some ppl. Maybe we should make the opt-in more clear.
What about providing a new `cache.adapter.redis_tag_aware` base adapter?